### DISCUSSION TITLE
How to merge separate animations to one?

### DISCUSSION THREAD

**Discussion by meightythree** [donmccurdy/glTF-Transform#732] [Q&A] [answered]
I have got multiple animations in the same glb under animation groups, how can I merge them to one?
![image](https://user-images.githubusercontent.com/23060960/202856457-d3b49140-1a30-4c44-8dcf-0c9ea171e63c.png)

**Comment by donmccurdy** [comment, marked as answer]
Hi @meightythree! I'll give the short answer here, but note that it depends on the animations being _compatible_. If you have two animations that each attempt to specify the position of every joint in the same character armature, merging them is going to be more complicated and I might suggest using Blender instead. But assuming the animations target different parts of the scene, you can merge them as follows:


```javascript
import { dedup, prune } from '@gltf-transform/functions';

const root = document.getRoot();
const mainAnim = root.listAnimations()[0];

for (const anim of root.listAnimations()) {
  if (anim === mainAnim) continue;
  
  // add channels and samplers to main animation
  for (const channel of anim.listChannels()) {
    mainAnim.addChannel(channel);
  }
  for (const sampler of anim.listSamplers()) {
    mainAnim.addSampler(sampler);
  }

  // remove old animation
  anim.dispose();
}

// clean up remaining resources
await document.transform(prune(), dedup());
```

You can try this in https://gltf.report/ quickly, without having to write the I/O code.

**Reply by meightythree** [reply]
It works, thank you @donmccurdy!

**Comment by Ruth678** [comment]
Nice information[,](https://docs.google.com/spreadsheets/d/1iGk8TKsiDziC5ifZyaVh_HSYvh8IqyBXd2JH2zPz4iU/edit#gid=0) I really like this.
